sale_substate/tests/test_sale_substate.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Akretion Mourad EL HADJ MIMOUNE
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+
+@tagged("-at_install", "post_install", "with_demo")
+class TestBaseSubstate(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.substate_test_sale = self.env["sale.order"]
+        self.substate_test_sale_line = self.env["sale.order.line"]
+
+        self.substate_under_nego = self.env.ref(
+            "sale_substate.base_substate_under_nego"
+        )
+        self.substate_won = self.env.ref("sale_substate.base_substate_won")
+        self.substate_wait_docs = self.env.ref("sale_substate.base_substate_wait_docs")
+        self.substate_valid_docs = self.env.ref(
+            "sale_substate.base_substate_valid_docs"
+        )
+        self.substate_in_delivery = self.env.ref(
+            "sale_substate.base_substate_in_delivery"
+        )
+        self.product_1 = self.env["product.product"].create(
+            {
+                "name": "Test Product 1",
+                "type": "service",
+            }
+        )
+
+    def test_sale_order_substate(self):
+        partner = self.env["res.partner"].create(
+            {
+                "name": "Test Partner",
+                "email": "test@example.com",
+            }
+        )
+        so_test1 = self.substate_test_sale.create(
+            {
+                "name": "Test base substate to basic sale",
+                "partner_id": partner.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_1.id,
+                            "product_uom_qty": 2,
+                            "product_uom": self.product_1.uom_id.id,
```

Comment:
```suggestion
                            "product_uom_id": self.product_1.uom_id.id,
```
It looks product_uom has been renamed in 19.0